deeplake/client/client.py

```diff
@@ -527,3 +529,113 @@ def has_indra_org_permission(self, org_id: str) -> Dict[str, Any]:
         ).json()
 
         return response
+
+
+class DeepMemoryBackendClient(DeepLakeBackendClient):
+    def __init__(self, token: Optional[str] = None):
+        super().__init__(token=token)
+
+    def deepmemory_is_available(self, org_id: str):
+        """Checks if DeepMemory is available for the user.
+        Args:
+            org_id (str): The name of the user/organization to which the dataset belongs.
+        Returns:
+            bool: True if DeepMemory is available, False otherwise.
+        """
+        response = self.request(
+            "GET",
+            f"/api/organizations/{org_id}/features/deepmemory",
+            endpoint=self.endpoint(),
+        )
+        return response.json()["available"]
+
+    def start_taining(
+        self,
+        corpus_path: str,
+        queries_path: str,
+    ) -> Dict[str, Any]:
+        """Starts training of DeepMemory model.
+        Args:
+            corpus_path (str): The path to the corpus dataset.
+            queries_path (str): The path to the queries dataset.
+        Returns:
+            Dict[str, Any]: The json response containing job_id.
+        """
+        response = self.request(
+            method="POST",
+            relative_url="/api/deepmemory/v1/train",
+            json={"corpus_dataset": corpus_path, "query_dataset": queries_path},
+        )
+        check_response_status(response)
+        return response.json()
+
+    def cancel_job(self, job_id: str):
+        """Cancels a job with job_id.
+        Args:
+            job_id (str): The job_id of the job to be cancelled.
+        Returns:
+            bool: True if job was cancelled successfully, False otherwise.
+        """
+        try:
+            response = self.request(
+                method="POST",
+                relative_url=f"/api/deepmemory/v1/jobs/{job_id}/cancel",
+            )
+        except Exception as e:
+            print(f"Job with job_id='{job_id}' was not cancelled!\n Error: {e}")
+            return False
+
+        check_response_status(response)
```

Comment:
`check_response_status(response)` also rethrows some exceptions, based on API Response status as far as I understand. Should it be moved up into the try-except, or is it left here intentionally?